extensions/amp-carousel/0.1/slides.js

```diff
@@ -45,6 +55,16 @@ export class AmpSlides extends BaseCarousel {
 
     /** @private {number} */
     this.currentIndex_ = 0;
+
+    /** @private {?number} */
+    this.autoplayTimeoutId_ = null;
+
+    this.setupAutoplay_();
+
+    // Setup microtask to let the runtime settle.
+    // TODO: (erwinm) experiment with only letting autoplay run if the
+    // carouse is in the viewport.
+    timer.delay(this.tryAutoplay_.bind(this , 1, true), 0);
```

Comment:
How about setting the timer in `#setupAutoplay_`? It'll avoid setting up this timer when there's no `autoplay` attribute, and guard against non-positive values.